src/ILLink.Tasks/LinkTask.cs

```diff
@@ -51,11 +51,24 @@ public class ILLink : ToolTask
 
 		/// <summary>
 		///   The directory in which to place linked assemblies.
-		//    Maps to '-out'.
+		///    Maps to '-out'.
 		/// </summary>
 		[Required]
 		public ITaskItem OutputDirectory { get; set; }
 
+		/// <summary>
+		/// Turns off trim correctness analysis warnings.
+		/// Maps to '--nowarn 2006;2026;2037;2043'.
+		/// </summary>
+		public bool TrimmerAnalysisWarnings { set => _trimmerAnalysisWarnings = value; }
+		bool? _trimmerAnalysisWarnings;
```

Comment:
I just wanted to start the discussion where should the list of "trimmer analysis warnings" be defined.
Possibilities:
* linker itself
* ILLink task
* ILLink target

I personally don't want to put it into linker itself. But I'm split on whether it should be in the task or the target. Target gives us potentially greater flexibility with regard to versioning and so on. The task is better since it lives in the same repo as the linker.

@sbomer any suggestions?

Reply:
My understanding of the conclusion (https://github.com/mono/linker/pull/1303#issuecomment-653433498) was that we were only going to add an option for `--nowarn`, and not to design the option for disabling categories yet. They would be disabled from the MSBuild targets via the `NoWarn` property, not in the task itself.

If we were to introduce a warning category option, I think it should go in the linker, since I have been thinking of it as enabling or disabling different "linker analyzers". For versioning, we could always pass a TFM from the SDK into the linker, which the linker can use in its decisions about what to output.